src/js/components/Anchor/index.d.ts

```diff
@@ -13,7 +13,7 @@ export interface AnchorProps {
   onClick?: ((...args: any[]) => any);
   reverse?: boolean;
   size?: "xsmall" | "small" | "medium" | "large" | "xlarge" | "xxlarge" | string;
-  as?: string;
+  as?: string | React.Component | React.FC | ((...args: any[]) => any);
```

Comment:
i am wondering whether we should align with the typings of styled-components for this.
`keyof JSX.IntrinsicElements | React.ComponentType<any>`
both comments concern all the typings change.